File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace EUD_ShowRibbonForm {
    public partial class Form1: Form {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            // It's required to enable form skins in the application.
            DevExpress.Skins.SkinManager.EnableFormSkins();
            DevExpress.LookAndFeel.LookAndFeelHelper.ForceDefaultLookAndFeelChanged();
        }

        // Show XRDesignRibbonFormEx.
        private void button1_Click(object sender, EventArgs e) {
            // Create an End-User Designer Form with a Ribbon.
            XRDesignRibbonFormEx form = new XRDesignRibbonFormEx();

            // Create a new blank report.
            form.OpenReport(new XtraReport());

            // Show the End-User Designer Form with a Ribbon
            form.Show();
        }
        
        // Show XRDesignRibbonFormEx modally.
        private void button2_Click(object sender, EventArgs e) {
            // Create an End-User Designer Form with a Ribbon.
            XRDesignRibbonFormEx form = new XRDesignRibbonFormEx();

            // Create a new blank report.
            form.OpenReport(new XtraReport());

            // Show the End-User Designer Form with a Ribbon
            form.ShowDialog();
        }

        // Call the ShowRibbonDesigner() method for the existing report.
        private void button3_Click(object sender, EventArgs e) {
            ReportDesignTool dt = new ReportDesignTool(new XtraReport1());
            dt.ShowRibbonDesigner();
        }
        
        // Call ShowRibbonDesignerDialog() method for the existing report.
        private void button4_Click(object sender, EventArgs e) {
            ReportDesignTool dt = new ReportDesignTool(new XtraReport1());
            dt.ShowRibbonDesignerDialog();
        }

    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace EUD_ShowRibbonForm
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' It's required to enable form skins in the application.
			DevExpress.Skins.SkinManager.EnableFormSkins()
			DevExpress.LookAndFeel.LookAndFeelHelper.ForceDefaultLookAndFeelChanged()
		End Sub

		' Show a report in a Ribbon End-User Designer form.
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			CType(New XtraReport1(), XtraReport1).ShowRibbonDesigner()
		End Sub

		' Show a report in a Ribbon End-User Designer form, modally.
		Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
			CType(New XtraReport1(), XtraReport1).ShowRibbonDesignerDialog()
		End Sub

		' Use a ReportDesignTool, to show a report in a Ribbon End-User Designer form.
		Private Sub button3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button3.Click
			Dim dt As New ReportDesignTool(New XtraReport1())
			dt.ShowRibbonDesigner()
		End Sub

		' Use a ReportDesignTool, to show a report in a Ribbon End-User Designer form.
		Private Sub button4_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button4.Click
			Dim dt As New ReportDesignTool(New XtraReport1())
			dt.ShowRibbonDesignerDialog()
		End Sub

		' Show an XRDesignRibbonForm.
		Private Sub button5_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button5.Click
			Dim form As New XRDesignRibbonForm()
			form.OpenReport(New XtraReport())
			form.Show()
		End Sub

		' Show an XRDesignRibbonForm, modally.
		Private Sub button6_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button6.Click
			Dim form As New XRDesignRibbonForm()
			form.OpenReport(New XtraReport())
			form.ShowDialog()
		End Sub

	End Class
End Namespace